api/tasks.go

```diff
@@ -639,16 +639,21 @@ func (g *TaskGroup) AddSpread(s *Spread) *TaskGroup {
 
 // LogConfig provides configuration for log rotation
 type LogConfig struct {
-	MaxFiles      *int  `mapstructure:"max_files" hcl:"max_files,optional"`
-	MaxFileSizeMB *int  `mapstructure:"max_file_size" hcl:"max_file_size,optional"`
-	Enabled       *bool `mapstructure:"enabled" hcl:"enabled,optional"`
+	MaxFiles      *int `mapstructure:"max_files" hcl:"max_files,optional"`
+	MaxFileSizeMB *int `mapstructure:"max_file_size" hcl:"max_file_size,optional"`
+
+	// COMPAT(1.7.0): Enabled had to be swapped for Disabled to fix a backwards
+	// compatibility bug when restoring pre-1.5.4 jobs. Remove in 1.7.0
+	Enabled *bool `mapstructure:"enabled" hcl:"enabled,optional"`
```

Comment:
Can we remove in 1.6.0? I see this as less of a traditional deprecation and more of a bug removal that unfortunately needs to span >1 release.

I think it's safe to remove in 1.6.0 because:

1. Basically everyone who upgraded to 1.5.4 *must upgrade to 1.5.5* and therefore won't need this field in 1.6.0.
2. Anyone who didn't upgrade to 1.5.4 won't need this compat field in 1.6.0 anyway

(if somehow there are people who upgrade to 1.5.4 and don't care either way... i don't think they'll care about when this is removed either)

Reply:
Agreed. Will fix.